src/client-library/deriv-api-client-legacy.ts

```diff
@@ -7,16 +7,48 @@ type WebsocketMap = Map<string, typeof DerivAPI>;
 export class DerivAPILegacy {
     activeSocket: string;
     websocketMap: WebsocketMap = new Map();
+    keepAliveIntervalId: NodeJS.Timeout | null = null;
 
     constructor() {
         const websocketURL = URLUtils.getWebsocketURL();
         const webSocketInstance = new WebSocket(websocketURL);
         const derivAPI = new DerivAPI({ connection: webSocketInstance });
         this.websocketMap.set(websocketURL, derivAPI);
         this.activeSocket = websocketURL;
+
+        this.keepAlive();
     }
 
     getActiveSocket() {
         return this.websocketMap.get(this.activeSocket);
     }
+
+    handleReconnect() {
+        const activeSocket = this.getActiveSocket();
+
+        if (!activeSocket || activeSocket.connection.readyState === WebSocket.CLOSED) {
+            const websocketURL = URLUtils.getWebsocketURL();
+            const webSocketInstance = new WebSocket(websocketURL);
+            const derivAPI = new DerivAPI({ connection: webSocketInstance });
+            this.websocketMap.set(websocketURL, derivAPI);
+            this.activeSocket = websocketURL;
```

Comment:
Also, this stuff seems to be repeated in the constructor, is it possible to create a helper method for them?
```suggestion
        if (!activeSocket || activeSocket.connection.readyState === WebSocket.CLOSED) {
            const websocketURL = URLUtils.getWebsocketURL();
            const webSocketInstance = new WebSocket(websocketURL);
            const derivAPI = new DerivAPI({ connection: webSocketInstance });
            this.websocketMap.set(websocketURL, derivAPI);
            this.activeSocket = websocketURL;
```

Reply:
can do, will refactor